### PR TITLE
A couple bug fixes

### PR DESCRIPTION
**Pull request by dylan-conway**

### CODE REVIEW

File: test/js/node/fs/fs.test.ts [hunk 1]
```diff
@@ -677,6 +677,29 @@ describe("createReadStream", () => {
       });
     });
   });
+
+  it("works with very large file", async () => {
+    const tempFile = tmpdir() + "/" + "large-file" + Date.now() + ".txt";
+    Bun.write(Bun.file(tempFile), "big data big data big data".repeat(10000));
```
Comment (Jarred-Sumner) [inline test/js/node/fs/fs.test.ts]: needs an await

File: src/bun.js/bindings/bindings.cpp [hunk 2]
```diff
@@ -2834,9 +2832,23 @@ int32_t JSC__JSValue__toInt32(JSC__JSValue JSValue0)
...
+}
+
+int64_t JSC__JSValue__coerceToInt64(JSC__JSValue JSValue0, JSC__JSGlobalObject* arg1)
+{
+    JSValue value = JSValue::decode(JSValue0);
+    if (value.isCell() && value.isHeapBigInt()) {
+        return value.toBigInt64(arg1);
+    }
+
     return value.toInt32(arg1);
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/bindings.cpp:2857]: tryConvertToint52

Comment (Jarred-Sumner) [reply src/bun.js/bindings/bindings.cpp:2857]: we basically already do this for toInt64

Comment (dylan-conway) [reply src/bun.js/bindings/bindings.cpp:2857]: added 👍

File: src/bun.js/webcore/body.zig [hunk 3]
```diff
@@ -174,23 +174,13 @@ pub const Body = struct {
...
-
-                    if (less_than and greater_than) {
-                        result.status_code = @truncate(u16, @intCast(u64, status_value.toInt64()));
-                    }
-                } else if (status_value.isNumber()) {
-                    const number = status_value.to(i32);
-                    if (100 <= number and number < 999)
-                        result.status_code = @truncate(u16, @intCast(u32, number));
+                const number = status_value.coerceToInt64(ctx);
+                if ((200 <= number and number < 600) or number == 101) {
```
Comment (Electroid) [inline src/bun.js/webcore/body.zig:178]: We should technically allow up to `999`

Comment (Electroid) [reply src/bun.js/webcore/body.zig:178]:
```
❯ curl -I --url http://www.linkedin.com/in/blahblah
HTTP/1.1 999 Request denied
```

**Pull request merged**